mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -558,19 +560,19 @@ unsafe extern "C" fn meta_request_headers_callback(
 }
 
 /// SAFETY: Don't call this function directly, only called by the CRT as a callback.
-unsafe extern "C" fn meta_request_receive_body_callback(
+unsafe extern "C" fn meta_request_receive_body_callback_ex(
     _request: *mut aws_s3_meta_request,
     body: *const aws_byte_cursor,
-    range_start: u64,
+    meta: aws_s3_meta_request_receive_body_extra_info,
     user_data: *mut libc::c_void,
 ) -> i32 {
     // SAFETY: user_data always will be a MetaRequestOptionsInner since that's what we set it to
     // in MetaRequestOptions::new.
     let user_data = MetaRequestOptionsInner::from_user_data_ptr(user_data);
 
-    if let Some(callback) = user_data.on_body.as_mut() {
-        let slice: &[u8] = aws_byte_cursor_as_slice(&*body);
-        callback(range_start, slice);
+    if let Some(callback) = user_data.on_body_ex.as_mut() {
+        let buffer = Buffer::new_unchecked(&*body, &meta.ticket);
```

Comment:
Ideally should have a safety comment saying that `buffer` is dropped before `body` and `meta.ticket`